packages/pages/src/common/src/template/internal/getTemplateFilepaths.ts

```diff
@@ -38,3 +49,89 @@ export const getTemplateFilepathsFromProjectStructure = (
       : [projectStructure.templatesRoot]
   );
 };
+
+const globalClientRenderFilename = "_client.tsx";
+const globalServerRenderFilename = "_server.tsx";
+
+/**
+ * Determines the client and server rendering templates to use. It first looks for a _client/server.tsx file in the scoped
+ * path. Then looks in the templatesRoot path. Finally, if none exist, defaults to a built-in rendering template that aligns
+ * with the originally hardcoded template that was used before this configurable feature was added.
+ *
+ * This DOES NOT take into account template-level rendering customizability. It only returns the global render templates.
+ *
+ * @param templatesRootPath the path where the templates live, typically src/templates
+ * @param scopedTemplatePath the subfolder path inside templatesRoot to scope to - used in multibrand setups
+ */
+export const getGlobalClientServerRenderTemplates = (
+  templatesRootPath: Path,
+  scopedTemplatePath: Path | undefined
+): ClientServerRenderTemplates => {
+  const [clientRenderTemplatePath, usingCustomClient] = findGlobalRenderFile(
+    templatesRootPath,
+    scopedTemplatePath,
+    globalClientRenderFilename
+  );
+  const [serverRenderTemplatePath, usingCustomServer] = findGlobalRenderFile(
+    templatesRootPath,
+    scopedTemplatePath,
+    globalServerRenderFilename
+  );
+
+  return {
+    clientRenderTemplatePath,
+    serverRenderTemplatePath,
+    isCustomRenderTemplate: usingCustomClient || usingCustomServer,
+  };
+};
+
+/**
+ * @param templatesRootPath the path where the templates live, typically src/templates
+ * @param scopedTemplatePath the subfolder path inside templatesRoot to scope to - used in multibrand setups
+ * @param globalFilename the file to find
+ * @returns the path to the appropriate file along with a boolean denoting if the path returned is a custom render template
+ */
+const findGlobalRenderFile = (
+  templatesRootPath: Path,
+  scopedTemplatePath: Path | undefined,
+  globalFilename: string
+): [string, boolean] => {
```

Comment:
it'd be more clear if you read the file right away and passed around a file object rather than the absolute path to the file

Reply:
I need the path vs the contents at times. For instance when we get the hydration template:
```
import {render} from "${convertToPosixPath(clientRenderTemplatePath)}";
```

and in the dev server:
```
const serverRenderTemplateModule = (await vite.ssrLoadModule(
    clientServerRenderTemplates.serverRenderTemplatePath
  )) as RenderTemplate;
```

Are you saying to load the file as a module directly here, but return some type that has other things, like the path and isCustom?
```
interface RenderTemplateInternal {
    module: RenderTemplate;
    path: string;
    isCustom: boolean;
}
```

Reply:
Yeah something along those lines.